FirebaseRemoteConfig/SwiftNew/ConfigFetch.swift

```diff
@@ -0,0 +1,814 @@
+            status: strongSelf.settings.lastFetchStatus,
+            error: error
+          )
+        return
+      }
+      strongSelf.settings.isFetchInProgress = true
+      let fetchTypeHeader = "\(baseFetchType)/1"
+      strongSelf
+        .refreshInstallationsToken(
+          withFetchHeader: fetchTypeHeader,
+          completionHandler: completionHandler,
+          updateCompletionHandler: nil
+        )
+    }
+  }
+
+  // MARK: - Fetch Helpers
+
+  /// Fetches config data immediately, keyed by namespace. Completion block will be called on the
+  /// main
+  /// queue.
+  /// - Parameters:
+  ///   - fetchAttemptNumber: The number of the fetch attempt.
+  ///   - completionHandler: Callback handler.
+  @objc public func realtimeFetchConfigWithNoExpirationDuration(_ fetchAttemptNumber: Int,
+                                                                completionHandler: @escaping (RemoteConfigFetchStatus,
+                                                                                              RemoteConfigUpdate?,
+                                                                                              Error?)
+                                                                  -> Void) {
+    // Note: We expect the googleAppID to always be available.
+    let hasDeviceContextChanged = Device.remoteConfigHasDeviceContextChanged(
+      settings.deviceContext,
+      projectIdentifier: options.googleAppID
+    )
+
+    lockQueue.async { [weak self] in
+      guard let strongSelf = self else { return }
+      // Check whether cache data is within throttle limit.
+      if strongSelf.settings.shouldThrottle() && !hasDeviceContextChanged {
+        // Must set lastFetchStatus before FailReason.
+        strongSelf.settings.lastFetchStatus = .throttled
+        strongSelf.settings.lastFetchError = .throttled
+        let throttledEndTime = strongSelf.settings.exponentialBackoffThrottleEndTime
+
+        let error = NSError(
+          domain: RemoteConfigErrorDomain,
+          code: RemoteConfigError.throttled.rawValue,
+          userInfo: [throttledEndTimeInSecondsKey: throttledEndTime]
+        )
+        strongSelf
+          .reportCompletion(
+            status: .failure,
+            update: nil,
+            error: error,
+            completionHandler: nil,
+            updateCompletionHandler: completionHandler
+          )
+        return
+      }
+      strongSelf.settings.isFetchInProgress = true
+
+      let fetchTypeHeader = "\(realtimeFetchType)/\(fetchAttemptNumber)"
+      strongSelf
+        .refreshInstallationsToken(
+          withFetchHeader: fetchTypeHeader,
+          completionHandler: nil,
+          updateCompletionHandler: completionHandler
+        )
+    }
+  }
+
+  /// Refresh installation ID token before fetching config. installation ID is now mandatory for
+  /// fetch
+  /// requests to work.(b/14751422).
+  private func refreshInstallationsToken(withFetchHeader fetchTypeHeader: String,
```

Comment:
This would be more readable starting the params on the next line.

Reply:
The style script keeps overwriting my changes here.

Reply:
Should I look into opting the file out of the style enforcement?

Reply:
Not yet. I'd rather do some experiments with inputs and options first. I'm fine to defer this for now.

Reply:
SG, thanks!